server/schedulers/hot_region.go

```diff
@@ -393,10 +393,23 @@ func (h *balanceHotRegionsScheduler) balanceByPeer(cluster opt.Cluster, storesSt
 		if srcStore == nil {
 			log.Error("failed to get the source store", zap.Uint64("store-id", srcStoreID))
 		}
+
+		srcPeer := srcRegion.GetStorePeer(srcStoreID)
+		if srcPeer == nil {
+			return nil, nil, nil, Influence{}
+		}
```

Comment:
why need to judge it in hot region rather than other schedulers?

Reply:
Why not use `continue` to just skip this peer?

Reply:
@shafreeck good idea.

Reply:
@lhy1024 because hot region scheduler selects region from stats, which may be out of date. Others do not have this problem.

Reply:
@lhy1024 For shcedulers that include label related tests (hotWrite and shuffleHot), I have changed them to test twice, the second time run with placement rules enabled.